src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,52 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int START_ARRAY_CAPACITY = 10;
+    private static final double CAPACITY_ARRAY_MULTIPLE = 1.5;
+    private int size;
+    private T[] stackArray;
+    private int capacity;
 
+    public MateStack() {
+        capacity = START_ARRAY_CAPACITY;
+        stackArray = (T[]) new Object[capacity];
     }
```

Comment:
try to rewrite this task using class Node

Reply:
It also good solution, but in my opinion when we use class Node it is easier. As you wish

Reply:
done